routes/index.js

```diff
@@ -9,16 +9,56 @@ var config = require('../app/config');
 
 var router = express.Router();
 
-/* GET / - redirect to notebook list page */
+/* GET / - index notebook or list of notebooks */
 router.get('/', function(req, res) {
-    res.redirect('/notebooks');
+    nbstore.getNotebooks().then(
+        function success(notebooks) {
+            //if notebook index exists redirect to it immediately
+            var notebookIndex = "index.ipynb"
+            //ensure notebook index search is case insensitive
+            var indexFound = -1;
+            for(i in notebooks) {
```

Comment:
Isn't `notebooks` an array? Typically `for..in` is used to iterate keys in an object. I would prefer either a `forEach` for its expressiveness or a normal `for` loop here if it is an array.

Additionally, it looks like you are creating an global variable `i`. I don't see any `var` declaration.